core/src/main/scala/org/apache/spark/sql/rapids/tool/store/TaskModel.scala

```diff
@@ -16,102 +16,155 @@
+
   def apply(event: SparkListenerTaskEnd): TaskModel = {
+    val taskInfo = event.taskInfo
+    val metrics = event.taskMetrics
+    val shuffleRead = metrics.shuffleReadMetrics
+    val shuffleWrite = metrics.shuffleWriteMetrics
+    val input = metrics.inputMetrics
+    val output = metrics.outputMetrics
     val reason = event.reason match {
       case failed: TaskFailedReason =>
         failed.toErrorString
       case _ =>
         event.reason.toString
     }
 
-    TaskModel(
-      event.stageId,
-      event.stageAttemptId,
-      event.taskType,
-      reason,
-      event.taskInfo.taskId,
-      event.taskInfo.attemptNumber,
-      event.taskInfo.launchTime,
-      event.taskInfo.finishTime,
-      event.taskInfo.duration,
-      event.taskInfo.successful,
-      event.taskInfo.executorId,
-      event.taskInfo.host,
-      event.taskInfo.taskLocality.toString,
-      event.taskInfo.speculative,
-      event.taskInfo.gettingResultTime,
-      event.taskMetrics.executorDeserializeTime,
-      event.taskMetrics.executorDeserializeCpuTime, // nanoseconds
-      event.taskMetrics.executorRunTime, // milliseconds
-      event.taskMetrics.executorCpuTime, // nanoseconds
-      event.taskMetrics.peakExecutionMemory,
-      event.taskMetrics.resultSize,
-      event.taskMetrics.jvmGCTime,
-      event.taskMetrics.resultSerializationTime,
-      event.taskMetrics.memoryBytesSpilled,
-      event.taskMetrics.diskBytesSpilled,
-      event.taskMetrics.shuffleReadMetrics.remoteBlocksFetched,
-      event.taskMetrics.shuffleReadMetrics.localBlocksFetched,
-      event.taskMetrics.shuffleReadMetrics.fetchWaitTime,
-      event.taskMetrics.shuffleReadMetrics.remoteBytesRead,
-      event.taskMetrics.shuffleReadMetrics.remoteBytesReadToDisk,
-      event.taskMetrics.shuffleReadMetrics.localBytesRead,
-      event.taskMetrics.shuffleReadMetrics.totalBytesRead,
-      event.taskMetrics.shuffleWriteMetrics.bytesWritten,
-      event.taskMetrics.shuffleWriteMetrics.writeTime, // nanoseconds
-      event.taskMetrics.shuffleWriteMetrics.recordsWritten,
-      event.taskMetrics.inputMetrics.bytesRead,
-      event.taskMetrics.inputMetrics.recordsRead,
-      event.taskMetrics.outputMetrics.bytesWritten,
-      event.taskMetrics.outputMetrics.recordsWritten)
+
+    var taskMetrics = immutable.IntMap.empty[Long]
+
+    def storeIfNonZero(field: Int, value: Long): Unit =
+      if (value != 0) taskMetrics = taskMetrics.updated(field, value)
+    storeIfNonZero(LAUNCH_TIME, taskInfo.launchTime)
+    storeIfNonZero(FINISH_TIME, taskInfo.finishTime)
+    storeIfNonZero(DURATION, taskInfo.duration)
+    storeIfNonZero(GETTING_RESULT_TIME, taskInfo.gettingResultTime)
+    storeIfNonZero(EXEC_DESERIALIZE_TIME, metrics.executorDeserializeTime)
+    storeIfNonZero(EXEC_DESERIALIZE_CPU_TIME, metrics.executorDeserializeCpuTime)
+    storeIfNonZero(EXEC_RUN_TIME, metrics.executorRunTime)
+    storeIfNonZero(EXEC_CPU_TIME, metrics.executorCpuTime)
```

Comment:
nit: Can we retain some of the comments that were present in the original metrics? (e.g. nanoseconds)